src/ngx-gallery-action.component.ts

```diff
@@ -1,30 +1,35 @@
-import { ChangeDetectionStrategy, Component, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
-import { NgxGalleryPreviewComponent } from "./ngx-gallery-preview.component";
-import { NgxGalleryScrollOverviewComponent } from "./ngx-gallery-scroll-overview/ngx-gallery-scroll-overview.component";
+import { ChangeDetectionStrategy, Component, Input, Output, EventEmitter, ViewChild } from '@angular/core';
+import { NgxGalleryPreviewComponent } from './';
+import { ZoomPosition } from './ngx-gallery-scroll-overview/zoomPosition';
 
 @Component({
     selector: 'ngx-gallery-action',
     template: `
-        <i class="ngx-gallery-icon {{ icon }}" [class.ngx-gallery-icon-disabled]="disabled"
+        <div class="ngx-gallery-icon {{ icon }}" [class.ngx-gallery-icon-disabled]="disabled"
             aria-hidden="true"
             title="{{ titleText }}"
             (click)="handleClick($event)" >
-           <ngx-gallery-scroll-overview #scrollOverviewComponent [gallery]="gallery" *ngIf="rectangleScroll"> </ngx-gallery-scroll-overview>
-        </i>`,
+           <ngx-gallery-scroll-overview #scrollOverviewComponent (onZoomChanged)="zoomChanged()" [zoomPosition] = "zoomPosition" *ngIf="showScrollOverview"> </ngx-gallery-scroll-overview>
```

Comment:
Yeah, this is what I meant. To make event more use of EventEmitter you should pass in the event (and the event could be an object like `{before: ZoomPosition after: ZoomPosition}`, but this would be defined in the `ScrollOverviewComponent`). And switch the names: The event should be named `zoomChanged`, because that is what just happend and the responding function should be named `onZoomChanged`, because this is the reaction to the event. Result would be: `(zoomChanged)="onZoomChanged($event)"`